addons/crm/tests/test_crm_pls.py

```diff
@@ -448,3 +469,63 @@ def test_settings_pls_start_date(self):
         res_config_new = resConfig.new()
         self.assertEqual(Date.to_string(res_config_new.predictive_lead_scoring_start_date),
             str_date_8_days_ago, "If config param is not a valid date, date in settings should be set to 8 days before today")
+
+    @users('user_sales_manager')
+    def test_team_unlink(self):
+        """ Test that frequencies are sent to "no team" when unlinking a team
+        in order to avoid loosing too much informations. """
+        pls_team = self.env["crm.team"].browse(self.pls_team.ids)
+
+        # clean existing data
+        self.env["crm.lead.scoring.frequency"].sudo().search([('team_id', '=', False)]).unlink()
+
+        # existing no-team data
+        no_team = [
+            ('stage_id', '1', 20, 10),
+            ('stage_id', '2', 0.1, 0.1),
+            ('stage_id', '3', 10, 0),
+            ('country_id', '1', 10, 0.1),
+        ]
+        self.env["crm.lead.scoring.frequency"].sudo().create([
+            {'variable': variable, 'value': value,
+             'won_count': won_count, 'lost_count': lost_count,
+             'team_id': False,
+            } for variable, value, won_count, lost_count in no_team
+        ])
+
+        # add some frequencies to team to unlink
+        team = [
+            ('stage_id', '1', 20, 10),  # existing noteam
+            ('country_id', '1', 0.1, 10),  # existing noteam
+            ('country_id', '2', 0.1, 0),  # new but void
+            ('country_id', '3', 30, 30),  # new
+        ]
+        existing_plsteam = self.env["crm.lead.scoring.frequency"].sudo().create([
+            {'variable': variable, 'value': value,
+             'won_count': won_count, 'lost_count': lost_count,
+             'team_id': pls_team.id,
+            } for variable, value, won_count, lost_count in team
+        ])
+
+        pls_team.unlink()
+
+        final_noteam = [
+            ('stage_id', '1', 40, 20),
+            ('stage_id', '2', 0.1, 0.1),
+            ('stage_id', '3', 10, 0),
+            ('country_id', '1', 10, 10),
+            ('country_id', '3', 30, 30),
+
+        ]
+        self.assertEqual(
+            existing_plsteam.exists(), self.env["crm.lead.scoring.frequency"],
+            'Frequencies of unlinked teams should be unlinked (cascade)')
+        existing_noteam = self.env["crm.lead.scoring.frequency"].sudo().search([
+            ('team_id', '=', False),
+            ('variable', 'in', ['stage_id', 'country_id']),
+        ])
+        for frequency in existing_noteam:
+            stat = next(item for item in final_noteam if item[0] == frequency.variable and item[1] == frequency.value)
```

Comment:
Would be a bit more clear with a list of dict with clear values ('variable', 'value', 'won_count', ...) instead of a list of lists, but it's not a huge deal.

Or maybe use the same trick as when you create:
`for variable, value, won_count, lost_count in existing_noteam`